src/Uno.UWP/Networking/Connectivity/ConnectionProfile.Android.cs

```diff
@@ -30,21 +30,52 @@ private ConnectionProfile()
 		{
 			NetworkInformation.VerifyNetworkStateAccess();
 			_connectivityManager = (AndroidConnectivityManager)ContextHelper.Current.GetSystemService(Context.ConnectivityService);
-#pragma warning disable CS0618 // Type or member is obsolete
-			NetworkInfo info = _connectivityManager.ActiveNetworkInfo;
-			if (info?.IsConnected == true)
+
+			if (Android.OS.Build.VERSION.SdkInt > Android.OS.BuildVersionCodes.LollipopMr1)
 			{
-				IsWwanConnectionProfile = IsConnectionWwan(info.Type);
-				IsWlanConnectionProfile = IsConnectionWlan(info.Type);
+				var activeNetwork = _connectivityManager.ActiveNetwork;
+				if (activeNetwork != null)
+				{
+					var netCaps = _connectivityManager.GetNetworkCapabilities(activeNetwork);
+					if (netCaps != null)
+					{
+						if (netCaps.HasTransport(Android.Net.TransportType.Wifi))
+						{
+							IsWlanConnectionProfile = true;
+						}
+						if (netCaps.HasTransport(Android.Net.TransportType.Cellular))
+						{
+							IsWwanConnectionProfile = true;
+						}
+

```

Comment:
```suggestion
						else if (netCaps.HasTransport(Android.Net.TransportType.Cellular))
						{
							IsWwanConnectionProfile = true;
						}
						else 
						{
						   IsWwanConnectionProfile = false;
						}


```

Reply:
Almost :)  (it can be both Wwan and Wlan).
Ok. Done.